8Knot/pages/index/search_utils.py

```diff
@@ -0,0 +1,103 @@
+"""
+Search utilities for the 8Knot application.
+Provides improved search algorithms for the searchbar.
+"""
+from typing import List, Dict, Any
+import re
+from fuzzywuzzy import fuzz
+from fuzzywuzzy import process
+from rapidfuzz import process
+
+
+def fuzzy_search(query: str, options: List[Dict[str, Any]], threshold: float = 0.2) -> List[Dict[str, Any]]:
+    """
+    Perform fuzzy search on a list of options using fuzzywuzzy, case-insensitive.
+    Args:
+        query: Search query string
+        options: List of dictionaries with 'label' and 'value' keys
+        threshold: Minimum similarity score to include in results (0-1)
+    Returns:
+        List of matching options sorted by relevance
+    """
+    if not query:
+        return options
+
+    # For very short queries (1-2 chars), use a simpler and faster matching approach
```

Comment:
I'd recommend breaking this down into a couple of utility functions. It's a bit hard to read as is.